app/src/main/java/org/ole/planet/myplanet/services/sync/SyncManager.kt

```
@@ -251,6 +257,14 @@ class SyncManager @Inject constructor(
         create(context, R.mipmap.ic_launcher, "Syncing data", "Please wait...")
     }
 
+    private suspend fun pushCurrentUserShelf() {
+        try {
+            userRepository.getUserModel()?.let { userSyncRepository.uploadShelfData(it) }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)